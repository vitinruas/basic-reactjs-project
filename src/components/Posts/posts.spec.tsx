import { render, screen } from '@testing-library/react'
import { Posts } from '.'

const fakeProps = {
  posts: [
    {
      id: 1,
      title: 'any_title_1',
      body: 'any_body_1',
      photo: 'any_photo_1',
      userId: 1,
    },
    {
      id: 2,
      title: 'any_title_2',
      body: 'any_body_2',
      photo: 'any_photo_2',
      userId: 2,
    },
  ],
  isItOver: false,
  loadMorePosts: () => {},
}

describe('<Posts />', () => {
  test('Should render <Posts />', async () => {
    render(<Posts {...fakeProps} />)

    const titleArray = screen.getAllByRole('heading', { name: /title/i })
    const imgArray = screen.getAllByRole('img', { name: /any_title/i })
    const bodyArray = await screen.findAllByText(/body/i)

    expect(titleArray).toHaveLength(2)
    expect(imgArray).toHaveLength(2)
    expect(bodyArray).toHaveLength(2)

    expect(screen.getByRole('img', { name: /any_title_1/i })).toHaveAttribute('src', 'any_photo_1')
  })

  test('Should not render posts', async () => {
    render(<Posts loadMorePosts={fakeProps.loadMorePosts} isItOver={fakeProps.isItOver} />)
    expect(screen.queryByRole('heading', { name: 'any_title_1' })).not.toBeInTheDocument()
  })

  test('Should match snapshot', async () => {
    const { container } = render(<Posts {...fakeProps} />)
    expect(container['firstChild']).toMatchSnapshot()
  })
})
