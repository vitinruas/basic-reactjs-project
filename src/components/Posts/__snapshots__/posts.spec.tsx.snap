// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> Should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    1
    <picture
      class="post-image"
    >
      <img
        alt="any_title_1"
        src="any_photo_1"
      />
    </picture>
    <div
      class="post-content"
    >
      <h1>
        any_title_1
      </h1>
      <p>
        Description: 
        any_body_1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    2
    <picture
      class="post-image"
    >
      <img
        alt="any_title_2"
        src="any_photo_2"
      />
    </picture>
    <div
      class="post-content"
    >
      <h1>
        any_title_2
      </h1>
      <p>
        Description: 
        any_body_2
      </p>
    </div>
  </div>
  <button
    class="button"
  >
    Load more Posts
  </button>
</div>
`;

exports[`<Posts /> Should render <Posts /> 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    1
    <picture
      class="post-image"
    >
      <img
        alt="any_title_1"
        src="any_photo_1"
      />
    </picture>
    <div
      class="post-content"
    >
      <h1>
        any_title_1
      </h1>
      <p>
        Description: 
        any_body_1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    2
    <picture
      class="post-image"
    >
      <img
        alt="any_title_2"
        src="any_photo_2"
      />
    </picture>
    <div
      class="post-content"
    >
      <h1>
        any_title_2
      </h1>
      <p>
        Description: 
        any_body_2
      </p>
    </div>
  </div>
  <button
    class="button"
  >
    Load more Posts
  </button>
</div>
`;
